### PR TITLE
[proofs] [alethe] Alethe let binding

### PR DESCRIPTION
**Pull request by HanielB**
Adds a specific let binder for Alethe proofs. It is necessary because sharing is represented via the ":named" attribute of SMT-LIB, where the first occurrence (in a post-order visit) is wrapped in a attribute to declare a name (i.e., a fresh variable) and the next occurrences are replaced by that variable.

### CODE REVIEW

File: src/proof/alethe/alethe_printer.h [hunk 1]
```diff
@@ -18,12 +18,44 @@
...
 #define CVC4__PROOF__ALETHE_PROOF_PRINTER_H
 
+#include "printer/let_binding.h"
 #include "proof/proof_node.h"
 
 namespace cvc5::internal {
 
 namespace proof {
 
+/** The Alethe-specific let binder.
```
Comment (ajreynol) [inline src/proof/alethe/alethe_printer.h:28]: Can we have this in its own file?

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: src/proof/alethe/alethe_printer.h (hunk 1 above)
Comment (HanielB) [reply src/proof/alethe/alethe_printer.h]: OK, moved.

File: src/proof/alethe/alethe_post_processor.h [hunk 2]
```diff
@@ -13,8 +13,8 @@
  * The module for processing proof nodes into Alethe proof nodes
  */
 
-#ifndef CVC4__PROOF__ALETHE_PROOF_PROCESSOR_H
-#define CVC4__PROOF__ALETHE_PROOF_PROCESSOR_H
+#ifndef CVC5__PROOF__ALETHE_PROOF_PROCESSOR_H
```
Comment (ajreynol) [inline src/proof/alethe/alethe_post_processor.h:16]: Actually this is also missing an `ALETHE__`

File: src/proof/alethe/alethe_printer.h [hunk 3]
```diff
@@ -15,9 +15,10 @@
 
 #include "cvc5_private.h"
 
-#ifndef CVC4__PROOF__ALETHE_PROOF_PRINTER_H
-#define CVC4__PROOF__ALETHE_PROOF_PRINTER_H
+#ifndef CVC5__PROOF__ALETHE_PROOF_PRINTER_H
```
Comment (ajreynol) [inline src/proof/alethe/alethe_printer.h:18]: Ditto.

File: src/proof/alethe/alethe_proof_rule.h [hunk 4]
```diff
@@ -15,8 +15,8 @@
 
 #include "cvc5_private.h"
 
-#ifndef CVC4__PROOF__ALETHE_PROOF_RULE_H
-#define CVC4__PROOF__ALETHE_PROOF_RULE_H
+#ifndef CVC5__PROOF__ALETHE_PROOF_RULE_H
```
Comment (ajreynol) [inline src/proof/alethe/alethe_proof_rule.h:18]: Ditto.

**Review by ajreynol — APPROVED** [review]
One minor request, else LGTM

File: src/proof/alethe/alethe_post_processor.h (hunk 2 above)
Comment (HanielB) [reply src/proof/alethe/alethe_post_processor.h:16]: Ops. :) Fixed

**Pull request merged**